### PR TITLE
Feat(doctrine): create new writeListener which only use processor

### PR DESCRIPTION
**Pull request by samuelbigard**
| Q             | A
| ------------- | ---
| Branch?       | main 
| Type       |  Feature
| License       | MIT
| Doc PR        | api-platform/docs#...

### CODE REVIEW

File: src/Core/EventListener/WriteListener.php [hunk 1]
```diff
@@ -65,6 +65,7 @@ public function __construct($dataPersister, $iriConverter = null, $resourceMetad
 
         $this->resourceMetadataFactory = $resourceMetadataFactory;
         $this->resourceClassResolver = $resourceClassResolver;
+        trigger_deprecation('api-platform/core', '2.7', sprintf('The listener %s is deprecated and will be removed in 3.0', __CLASS__));
```
Comment (dunglas) [inline src/Core/EventListener/WriteListener.php:68]: You should also add a `@deprecated` annotation in the PHPDoc of the class.

File: src/Symfony/EventListener/WriteListener.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        ) {
+            return;
+        }
+
+        if (!$operation || ($operation->getExtraProperties()['is_legacy_resource_metadata'] ?? false)) {
+            return;
+        }
+
+        if (
+            !$this->resourceMetadataFactory instanceof ResourceMetadataCollectionFactoryInterface &&
```
Comment (dunglas) [inline src/Symfony/EventListener/WriteListener.php:80]: This type assertion always returns `true` isn't it?

File: src/Symfony/EventListener/WriteListener.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            (!$attributes['persist']
+                || $this->isOperationAttributeDisabled($attributes, self::OPERATION_ATTRIBUTE_KEY)
+            )) {
+            return;
+        }
+
+        $context = ['operation' => $operation, 'legacy_attributes' => $attributes + ['identifiers' => $operation->getIdentifiers(), 'has_composite_identifier' => $operation->getCompositeIdentifier()]];
+
+        if ($this->processor instanceof ProcessorInterface && !$this->processor->supports($controllerResult, $attributes['identifiers'], $operation->getName(), $context)) {
```
Comment (dunglas) [inline src/Symfony/EventListener/WriteListener.php:89]: Same here, `$this->processor instanceof ProcessorInterface` is always `true`.

**Review by dunglas — CHANGES_REQUESTED** [review]

**Comment by soyuka** [comment]
Thanks @samuelbigard

**Pull request merged**